In [1]:
import pandas as pd
from sklearn.experimental import enable_iterative_imputer
from sklearn.impute import IterativeImputer
from sklearn.preprocessing import OneHotEncoder
from sklearn.compose import ColumnTransformer
from sklearn.impute import KNNImputer
import numpy as np
pd.set_option('display.max_columns', None)


In [3]:
df = pd.read_csv('HR_CLEAN_DATA.csv')
df.head(1)

Employee_Number  Age  Age_Group  Attrition Business_Travel  Daily_Rate  \
0              162   51        NaN      False             NaN       684.0   

  Department  Distance_From_Home  Education Education_Field  \
0        NaN                   6          3             NaN   

   Environment_Satisfaction Gender  Hourly_Rate  Job_Involvement  Job_Level  \
0                         1      M           51                3          5   

              Job_Role  Job_Satisfaction Marital_Status  Monthly_Income  \
0   Research Director                  3            NaN         19537.0   

   Monthly_Rate  Num_Companies_Worked Over_18 Over_Time  Percent_Salary_Hike  \
0        6462.0                     7       Y     False                   13   

   Performance_Rating  Relationship_Satisfaction  Standard_Hours  \
0                   3                          3             NaN   

   Stock_Option_Level  Total_Working_Years  Training_Times_Last_Year  \
0                   0                99999                         5   

   Work_Life_Balance  Years_At_Company  Years_Since_Last_Promotion  \
0                  3                20                          15   

   Years_With_Curr_Manager  Date_Birth  Remote_Work  
0                       15        1972         True

Performance_Rating:

In [4]:
conteo2_NaN = (df['Performance_Rating'] == 99999).sum()
total_valores = len(df)
porcentaje_99999 = (conteo2_NaN / total_valores) * 100

print(f"El valor 99999 aparece en {conteo2_NaN} registros, lo que representa el {porcentaje_99999:.2f}% del total.")

El valor 99999 aparece en 178 registros, lo que representa el 12.05% del total.


In [5]:
df['Standard_Hours'].unique()

array([nan, 80.])

Standard_Hours:

In [6]:
conteo2_NaN = (df['Standard_Hours'].isnull()).sum()
total_valores2 = len(df)
porcentaje2_NaN = (conteo2_NaN / total_valores2) * 100

print(f"El valor nulo aparece en {conteo2_NaN} registros, lo que representa el {porcentaje2_NaN:.2f}% del total.")

El valor nulo aparece en 1096 registros, lo que representa el 74.20% del total.


Performance_Rating Mediana y Media:

In [7]:
df['Performance_Rating'].replace(99999, np.nan, inplace=True)
media_performance_rating = df['Performance_Rating'].mean()
mediana_performance_rating = df['Performance_Rating'].median()

print(f"Media de Performance_Rating: {media_performance_rating}")
print(f"Mediana de Performance_Rating: {mediana_performance_rating}")

Media de Performance_Rating: 3.1562740569668977
Mediana de Performance_Rating: 3.0


Standard Hours Mediana:

In [8]:
mean_standard_hours = df['Standard_Hours'].mean()
median_standard_hours = df['Standard_Hours'].median()

print(f"Media de Standard_Hours: {mean_standard_hours}")
print(f"Mediana de Standard_Hours: {median_standard_hours}")

Media de Standard_Hours: 80.0
Mediana de Standard_Hours: 80.0


Conclusion:

1. 'Standard_Hours' = borrar columna.
- El valor nan aparece en 1096 registros, lo que representa el 74.20% del total.
2. 'Performance_Rating' = Reemplazar con la Implementación de Imputación por Mediana: 3 
- El valor 99999 aparece en 178 registros, lo que representa el 12.05% del total.